pkg/autopilot/controller/root_worker.go

```diff
@@ -67,18 +70,25 @@ func (w *rootWorker) Run(ctx context.Context) error {
 		logger.WithError(err).Fatal("unable to register autopilot scheme")
 	}
 
-	if err := RegisterIndexers(ctx, mgr, "worker"); err != nil {
-		logger.WithError(err).Fatal("unable to register indexers")
-	}
-
-	if err := apsig.RegisterControllers(ctx, logger, mgr, apdel.NodeControllerDelegate(), w.cfg.K0sDataDir); err != nil {
-		logger.WithError(err).Fatal("unable to register 'controlnodes' controllers")
-	}
-
-	// The controller-runtime start blocks until the context is cancelled.
-	if err := mgr.Start(ctx); err != nil {
-		return fmt.Errorf("unable to run controller-runtime manager for workers: %w", err)
-	}
-
-	return nil
+	return retry.OnError(wait.Backoff{
```

Comment:
Can you add comments why this backoff is needed? ie. is there a race or dependency that may not be obvious to someone looking at the code.